packages/api-clients/open-api/authorizationApi.yml

```diff
@@ -2191,9 +2205,25 @@ components:
       required:
         - purposeId
     ProducerKeychain:
+      description: |
+        Models a producer keychain that can be either a full producer keychain or a partial producer keychain based on the caller
+        tenant. In case the caller tenant is the producer of the keychain (i.e., the owner), the
+        full keychain is returned, otherwise a partial keychain is returned.
+      oneOf:
+        - $ref: "#/components/schemas/FullProducerKeychain"
+        - $ref: "#/components/schemas/PartialProducerKeychain"
+      discriminator:
+        propertyName: visibility
+        mapping:
+          PARTIAL: "#/components/schemas/PartialProducerKeychain"
+          FULL: "#/components/schemas/FullProducerKeychain"
+    FullProducerKeychain:
```

Comment:
Just for consistency with Client, since you are updating the model, let's add both the following properties in `users` and `eservices` - as done in Client for `users` and `purposes`

```
          default: []
          uniqueItems: true
```